Libraries/PeriphDrivers/Source/I3C/i3c_reva.h

```diff
@@ -0,0 +1,273 @@
+typedef struct _i3c_reva_ccc_req_t mxc_i3c_reva_ccc_req_t;
+typedef struct _i3c_reva_req_t mxc_i3c_reva_req_t;
+
+/**
+ * @brief   The list of high-keeper options.
+ *
+ * This setting should match the high-keeper implementation of the device.
+ */
+typedef enum {
+    MXC_I3C_REVA_HIGH_KEEPER_OFF = MXC_S_I3C_REVA_CONT_CTRL0_HKEEP_OFF, ///< No high-keeper support
+    MXC_I3C_REVA_HIGH_KEEPER_ON_CHIP =
+        MXC_S_I3C_REVA_CONT_CTRL0_HKEEP_ON_CHIP, ///< SCL and SDA pads
+    ///< have weak pull-ups
+    MXC_I3C_REVA_HIGH_KEEPER_EXT_SDA =
+        MXC_S_I3C_REVA_CONT_CTRL0_HKEEP_EXT_SDA, ///< External high-keeper
+    ///< support for SDA signal
+    MXC_I3C_REVA_HIGH_KEEPER_EXT_SCL_SDA = MXC_S_I3C_REVA_CONT_CTRL0_HKEEP_EXT_SCL_SDA,
+    ///< External high-keeper support for SCL and SDA signals
+} mxc_i3c_reva_high_keeper_t;
+
+/**
+ * @brief   The list of receive FIFO trigger levels.
+ */
+typedef enum {
+    MXC_I3C_REVA_RX_TH_NOT_EMPTY = MXC_V_I3C_REVA_CONT_FIFOCTRL_RX_THD_LVL_NOT_EMPTY, ///<
+    MXC_I3C_REVA_RX_TH_QUARTER_FULL = MXC_V_I3C_REVA_CONT_FIFOCTRL_RX_THD_LVL_QUARTER_FULL, ///<
+    MXC_I3C_REVA_RX_TH_HALF_FULL = MXC_V_I3C_REVA_CONT_FIFOCTRL_RX_THD_LVL_HALF_FULL, ///<
+    MXC_I3C_REVA_RX_TH_3_QUARTER_FULL = MXC_V_I3C_REVA_CONT_FIFOCTRL_RX_THD_LVL_3_QUARTER_FULL, ///<
+} mxc_i3c_reva_rx_threshold_t;
+
+/**
+ * @brief   The list of transmit FIFO trigger levels.
+ */
+typedef enum {
+    MXC_I3C_REVA_TX_TH_EMPTY = MXC_V_I3C_REVA_CONT_FIFOCTRL_TX_THD_LVL_EMPTY, ///<
+    MXC_I3C_REVA_TX_TH_QUARTER_FULL = MXC_V_I3C_REVA_CONT_FIFOCTRL_TX_THD_LVL_QUARTER_FULL, ///<
+    MXC_I3C_REVA_TX_TH_HALF_FULL = MXC_V_I3C_REVA_CONT_FIFOCTRL_TX_THD_LVL_HALF_FULL, ///<
+    MXC_I3C_REVA_TX_TH_ALMOST_FULL = MXC_V_I3C_REVA_CONT_FIFOCTRL_TX_THD_LVL_ALMOST_FULL, ///<
+} mxc_i3c_reva_tx_threshold_t;
+
+/**
+ * @brief IBI types.
+ *
+ */
+typedef enum {
+    MXC_I3C_REVA_IBI_TYPE_NONE = MXC_V_I3C_REVA_CONT_STATUS_IBITYPE_NONE, ///<
+    MXC_I3C_REVA_IBI_TYPE_IBI = MXC_V_I3C_REVA_CONT_STATUS_IBITYPE_IBI, ///<
+    MXC_I3C_REVA_IBI_TYPE_CONTROLLER_REQ = MXC_V_I3C_REVA_CONT_STATUS_IBITYPE_CONT_REQ, ///<
+    MXC_I3C_REVA_IBI_TYPE_HOTJOIN_REQ = MXC_V_I3C_REVA_CONT_STATUS_IBITYPE_HOTJOIN_REQ, ///<
+} mxc_i3c_reva_ibi_type_t;
+
+/**
+ * @brief Common Command Code request structure.
+ *
+ * Broadcast and direct commands are distinguished by MSB, i.e. if MSB being set
+ * implies a direct CCC while MSB being 0 implies a broadcast CCC.
+ *
+ */
+struct _i3c_reva_ccc_req_t {
+    uint8_t ccc; ///< CCC command to send.
+    uint8_t target_addr : 7; ///< Target address if CCC is a direct command. Ignored
+    ///< if CCC is a broadcast command.
+    uint8_t read_write : 1; ///< Set to 1 for read commands and set to 0 for write
+    ///< commands. Ignored for broadcast CCCs.
+    uint8_t flags; ///< See request flags.
+    uint8_t def_byte; ///< Optional defining byte. Defined by CCC.
+    uint8_t subCmd; ///< Optional sub-command. Defined by CCC. Only used by direct
```

Comment:
Suggest `sub_cmd` to match style of other field names

Reply:
Done.